pkg/unittest/assertion.go

```diff
@@ -41,61 +41,63 @@ func (a *Assertion) Assert(
 	assertionPassed := false
 	failInfo := make([]string, 0)
 
-	// Sort or defaultTemplates to ensure a consistent output
-	sort.Strings(a.defaultTemplates)
-
-	for idx, template := range a.defaultTemplates {
-		rendered, ok := templatesResult[template]
-		var validatePassed bool
-		var singleFailInfo []string
-		if !ok && a.requireRenderSuccess {
-			noFile := []string{"Error:", a.noFileErrMessage(template)}
-			failInfo = append(failInfo, noFile...)
-			assertionPassed = false
-			break
-		}
+	selectedDocsByTemplate, indexError := a.selectDocumentsForAssertion(a.getDocumentsByDefaultTemplates(templatesResult))
+	selectedTemplates := a.getKeys(selectedDocsByTemplate)
 
-		if a.requireRenderSuccess != renderSucceed {
-			invalidRender := ""
-			if len(rendered) > 0 {
-				invalidRender = fmt.Sprintf("Error: Invalid rendering: %s", rendered[0][common.RAW])
-			} else {
-				invalidRender = "Error: rendered manifest is empty"
+	// Sort templates to ensure a consistent output
+	sort.Strings(selectedTemplates)
+
+	if indexError != nil {
+		invalidDocumentIndex := []string{"Error:", indexError.Error()}
+		failInfo = append(failInfo, invalidDocumentIndex...)
+	} else {
+		for idx, template := range selectedTemplates {
```

Comment:
We should iterate only over **selectedTemplates** - That what's makes skipEmptyTemplates flag work.